copyright fails ideaProject build with cryptic error message

I just ran into the issue when `./gradlew idea` yields this cryptic error message:

```
:ideaProject FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':ideaProject'.
> Cannot access last() element from an empty List

* Try:
Run with --info or --debug option to get more log output.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':ideaProject'.
    at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:69)
    at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:46)
    at org.gradle.api.internal.tasks.execution.PostExecutionAnalysisTaskExecuter.execute(PostExecutionAnalysisTaskExecuter.java:35)
    at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:68)
    at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
    at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:52)
    at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
    at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:53)
    at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
    at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:203)
    at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:185)
    at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.processTask(AbstractTaskPlanExecutor.java:66)
    at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.run(AbstractTaskPlanExecutor.java:50)
    at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:54)
    at org.gradle.internal.concurrent.StoppableExecutorImpl$1.run(StoppableExecutorImpl.java:40)
Caused by: java.util.NoSuchElementException: Cannot access last() element from an empty List
    at com.palantir.baseline.plugins.BaselineIdea.addCopyright(BaselineIdea.groovy:94)
    at com.palantir.baseline.plugins.BaselineIdea$_apply_closure1$_closure8.doCall(BaselineIdea.groovy:41)
    at org.gradle.api.internal.ClosureBackedAction.execute(ClosureBackedAction.java:67)
    at org.gradle.internal.xml.XmlTransformer$XmlProviderImpl.apply(XmlTransformer.java:165)
    at org.gradle.internal.xml.XmlTransformer.doTransform(XmlTransformer.java:137)
    at org.gradle.internal.xml.XmlTransformer.doTransform(XmlTransformer.java:129)
    at org.gradle.internal.xml.XmlTransformer.transform(XmlTransformer.java:109)
    at org.gradle.internal.xml.XmlTransformer$transform$0.call(Unknown Source)
    at org.gradle.plugins.ide.internal.generator.XmlPersistableConfigurationObject.store(XmlPersistableConfigurationObject.groovy:45)
    at org.gradle.plugins.ide.internal.generator.XmlPersistableConfigurationObject$store$2.callCurrent(Unknown Source)
    at org.gradle.plugins.ide.internal.generator.XmlPersistableConfigurationObject$store$2.callCurrent(Unknown Source)
    at org.gradle.plugins.ide.internal.generator.AbstractPersistableConfigurationObject.store(AbstractPersistableConfigurationObject.groovy:59)
    at org.gradle.plugins.ide.internal.generator.generator.PersistableConfigurationObjectGenerator.write(PersistableConfigurationObjectGenerator.java:42)
    at org.gradle.plugins.ide.internal.generator.generator.PersistableConfigurationObjectGenerator.write(PersistableConfigurationObjectGenerator.java:28)
    at org.gradle.plugins.ide.api.GeneratorTask.generate(GeneratorTask.java:81)
    at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:75)
    at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.doExecute(AnnotationProcessingTaskFactory.java:228)
    at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:221)
    at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:210)
    at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:585)
    at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:568)
    at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:80)
    at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:61)
    ... 14 more
```

Which fails on the following line:
https://github.com/palantir/gradle-baseline/blob/develop/gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineIdea.groovy#L94

```
def lastfileName = copyrightDir.relativize(copyrightFiles.iterator().toList().last().toPath())
```

Turns out my project lacked a .baseline/copyright/copyright-file.txt. Ideally, the procedure would give a clear error message before failing on the "last()" call describing that a .baseline/copyright folder is required

--- a/gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineIdea.groovy
+++ b/gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineIdea.groovy
@@ -22,6 +22,7 @@ import org.gradle.api.tasks.compile.JavaCompile
 import org.gradle.plugins.ide.idea.IdeaPlugin
 import org.gradle.plugins.ide.idea.model.IdeaModel
 
+import java.nio.file.Files
 import java.nio.file.Paths
 
 class BaselineIdea extends AbstractBaselinePlugin {
@@ -70,7 +71,9 @@ class BaselineIdea extends AbstractBaselinePlugin {
     private void addCopyright(node) {
         def copyrightManager = node.component.find { it.'@name' == 'CopyrightManager' }
         def copyrightDir = Paths.get("${configDir}/copyright/")
+        assert Files.exists(copyrightDir), "${copyrightDir} must exist"
         def copyrightFiles = project.fileTree(copyrightDir.toFile()).include("*")
+        assert copyrightFiles.iterator().hasNext(), "${copyrightDir} must contain one or more copyright file"
         copyrightFiles.each { File file ->
             def fileName = copyrightDir.relativize(file.toPath())
             def copyrightNode = copyrightManager.copyright.find {

